File: week#2/readability/readability.c
//Calculating readability tests with Coleman-Liau index formula.

#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
    char *input = get_string ("Text: "); //promp the user for the text.

    int c = strlen(input);
    double letters = 0;
    double words = 1;
    double sentences = 0;
    char lower[c];
    for (int i = 0; i < c; i++) //iterate every char of the inputed text.
    {
        lower[i] = tolower(input[i]);
        if (isalpha(lower[i])) //for counting the letters.
        {
            letters++;
        }
        if (isspace(lower[i]))
        {
            words++;
        }
        if (lower[i] == '!' || lower[i] == '.' || lower[i] == '?')
        {
            sentences++;
        }
    }
    double L = letters/words*100;
    double S = sentences/words*100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index >= 16)
    {
        printf ("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf ("Before Grade 1\n");
    }
    else
    {
        printf ("Grade %i\n", index);
    }
}

File: week#6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    length = len(text)
    print (f"length {length}")
    words = 1
    letters = 0
    sentences = 0
    end_sentences = ['!', '?', '.']
    for i in range(length):
        if text[i].lower().isalpha():
            letters += 1
        if text[i].isspace():
            words += 1
        if text[i] in end_sentences:
            sentences += 1

    L = letters / words*100
    S = sentences / words*100
    index = round((0.0588 * L)- (0.296 * S) - 15.8)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")

main()
